components/IframePreviewCard/card.tsx

```diff
@@ -18,19 +18,32 @@ const findOrCreatePortalContainer = () => {
   return portalContainer
 }
 
-const Portal = ({ children }) => {
+const Portal = ({ children }: { children: React.ReactNode }) => {
   return ReactDOM.createPortal(children, findOrCreatePortalContainer())
 }
 
+type IframePreviewCardProps = {
+  url: string
+  onIframeError?: Function
+  onClose: () => void
+  onMoveOrResize: (size: { x: number; y: number }) => void
+  defaultCoordinates: {
+    width: number
+    height: number
+    x: number
+    y: number
+  }
+}
+
 export const IframePreviewCard = ({
   url,
   onIframeError,
   onClose,
   onMoveOrResize,
   defaultCoordinates,
-}) => {
+}: IframePreviewCardProps) => {
   /** @type {React.RefObject<HTMLIFrameElement>} */
-  const iframeRef = React.useRef(null)
+  const iframeRef = React.useRef(null) as any // TODO: TS support
```

Comment:
Doen't `React.RefObject<HTMLIFrameElement>` work for you?

Reply:
Line 55 `const doc = iframe.contentDocument || iframe.contentWindow.document`
it can't work